src/components/Disputes/DisputeAutoReveal.js

```diff
@@ -0,0 +1,119 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { Button, GU, Info } from '@aragon/ui'
+import { useWallet } from '../../providers/Wallet'
+
+import { getAutoRevealRequest } from '../../services/autoReveal'
+import {
+  getAutoRevealPreference,
+  getCodeFromLocalStorage,
+  getOutcomeFromCommitment,
+} from '../../utils/crvoting-utils'
+
+function DisputeAutoReveal({ commitment, disputeId, onAutoReveal, roundId }) {
+  const { account } = useWallet()
+
+  const autoRevealRequested = useAutoRevealPolling(account, disputeId, roundId)
+
+  if (autoRevealRequested) {
+    return <Info>Auto reveal enabled!</Info>
+  }
+
+  return (
+    <RequestAutoReveal
+      commitment={commitment}
+      disputeId={disputeId}
+      onAutoReveal={onAutoReveal}
+      roundId={roundId}
+    />
+  )
+}
+
+function RequestAutoReveal({ commitment, disputeId, onAutoReveal, roundId }) {
+  const { account } = useWallet()
+  const autoRevealServiceEnabled = getAutoRevealPreference(account)
+
+  const handleSubmit = useCallback(
+    event => {
+      event.preventDefault()
+
+      const password = getCodeFromLocalStorage(account, disputeId)
+      const outcome = getOutcomeFromCommitment(commitment, password)
+
+      onAutoReveal(account, disputeId, roundId, outcome, password)
+    },
+    [account, commitment, disputeId, onAutoReveal, roundId]
+  )
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <Button
+        label="Reveal your vote with auto-reveal service"
+        mode="strong"
+        type="submit"
+        wide
+      />
+      {autoRevealServiceEnabled && (
+        <Info
+          mode="warning"
+          css={`
+            margin-top: ${2 * GU}px;
+          `}
+        >
+          The auto-reveal service request has failed. Please click on the button
+          above to resend the request, reveal your vote and enable the service.
+        </Info>
+      )}
+    </form>
+  )
+}
+
+function useAutoRevealPolling(account, disputeId, roundId) {
+  const [autoRevealRequested, setAutoRevealRequested] = useState(false)
+  const timer = 3000
+
+  // We start the timeout timer in 0 in order to immediately make the api call
```

Comment:
```suggestion
  // We start the timeout timer at 0 to immediately make the api call on render
```

But do we need to use a ref here? Couldn't we use a local variable in the closure? (E.g. we could provide as an argument to `fetchAutoReveal()`).

Reply:
True, we don't need a ref in this case.